tests/test_table_properties.rs

```diff
@@ -0,0 +1,200 @@
+// Copyright 2017 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
+use rocksdb::{DB, Range, Options, Writable, DBEntryType, TablePropertiesCollection,
+              TablePropertiesCollector, TablePropertiesCollectorFactory};
+use std::cmp::Ordering;
+use std::collections::HashMap;
+use std::fmt;
+use std::io::Cursor;
+use tempdir::TempDir;
+
+enum Props {
+    NumKeys = 0,
+    NumPuts,
+    NumMerges,
+    NumDeletes,
+}
+
+fn encode_u32(x: u32) -> Vec<u8> {
+    let mut w = Vec::new();
+    w.write_u32::<LittleEndian>(x).unwrap();
+    w
+}
+
+fn decode_u32(x: &[u8]) -> u32 {
+    let mut r = Cursor::new(x);
```

Comment:
`mut x: &[u8]` is not `x: &mut [u8]`. You can consider it as `mut x: Slice` and `x: MutSlice`.